src/openrct2/ride/TrackDesign.cpp

```diff
@@ -2111,7 +2112,9 @@ void track_design_draw_preview(TrackDesign* td6, uint8_t* pixels)
  */
 static map_backup* track_design_preview_backup_map()
```

Comment:
You will need to modify this so that the return is a std::unique_ptr<map_backup> as we need the map_backup to survive for the whole of the calling function (`track_design_draw_preview`)